crates/rustfix/src/replace.rs

```diff
@@ -97,96 +152,41 @@ impl Data {
             return Err(Error::DataLengthExceeded(range, self.original.len()));
         }
 
-        let insert_only = range.start == range.end;
-
-        // Since we error out when replacing an already replaced chunk of data,
-        // we can take some shortcuts here. For example, there can be no
-        // overlapping replacements -- we _always_ split a chunk of 'initial'
-        // data into three[^empty] parts, and there can't ever be two 'initial'
-        // parts touching.
-        //
-        // [^empty]: Leading and trailing ones might be empty if we replace
-        // the whole chunk. As an optimization and without loss of generality we
-        // don't add empty parts.
-        let Some(index_of_part_to_split) = self
-            .parts
-            .iter()
-            .position(|p| !p.data.is_inserted() && p.start <= range.start && p.end >= range.end)
-        else {
-            tracing::debug!(
-                "no single slice covering {}..{}, current slices: {:?}",
-                range.start,
-                range.end,
-                self.parts,
-            );
-            return Err(Error::MaybeAlreadyReplaced(range));
```

Comment:
If this is no longer needed, we should remove it (and maybe add `#[mon_exhaustive]` to the error enum as a follow-up).

Reply:
I agree that it should be removed, but _should_ it be marked `#[non_exhaustive]`? It seems like this variant was a bit of a hedge against an implementation error, but not really a mistake from the caller.